### PR TITLE
feat: Downloads by Platform plots

### PR DESCRIPTION
**Pull request by aliciaaevans**
A stacked bar plot showing downloads by os/arch. Once this is merged, I'll update bioconda-docs to add this to each package page.

Open to suggestions for color scheme if people associate certain colors by os (noarch gray?). This is just the default palette.

Example with fake data: (UPDATED)
visualization1

### CODE REVIEW

File: src/build_plots/common.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    repo = Repo("bioconda-stats")
+    tags = repo.tags
+
+    # for each package, get the most recent versions.tsv
+    package_count = 0
+    error_count = 0
+    for filename in os.listdir(
+        f"bioconda-stats/package-downloads/anaconda.org/bioconda/{category}"
+    ):
+        if filename.endswith(".tsv"):
```
Comment (epruesse) [inline src/build_plots/common.py:26]: You could instead filter the list to include only `*.tsv` files. Saves one level of indentation.

File: src/build_plots/common.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                        v.writelines(package_df.to_json(orient="records"))
+                        logger.debug(f"Saved data for {package} to {category}.json.")
+
+            except Exception as e:
+                # Log package name and continue with the rest
+                error_count += 1
+                e.args = (f"Error creating plot for {package}.",) + e.args
+                logger.exception(e)
+
+            finally:
```
Comment (epruesse) [inline src/build_plots/common.py:102]: You don't need a finally if you're catching all exceptions anyway.

File: src/build_plots/common.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+from ._vendor.conda.models.version import VersionOrder
+
+
+basicConfig(level=INFO)
+logger = getLogger(__name__)
+
+
+def buildDailyPlot(category, field, max_packages, days_to_plot):
+    if not os.path.exists("plots"):
+        os.makedirs("plots")
```
Comment (epruesse) [inline src/build_plots/common.py:15]: You can use `os.makedirs("plots", exists_ok = True)` to just have it ensure the dir is there.

File: src/build_plots/common.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    package_df[field] = pd.Categorical(
+                        package_df[field], ordered=True, categories=version_list
+                    )
+                    package_df = package_df[package_df[field].notna()].sort_values(
+                        by=[field, "date"]
+                    )[["date", "total", "delta", field]]
+
+                    # Save plot data
+                    if not os.path.exists(f"plots/{package}"):
+                        os.makedirs(f"plots/{package}")
```
Comment (epruesse) [inline src/build_plots/common.py:91]: Same as above `exists_ok` is your friend. :)

**Review by epruesse — COMMENTED** [review]
Looks nice :-)

Leaving approval to the others since I've been a bit inactive for a ... few years.

**Comment by daler** [comment]
Looks good. On one hand it might be interesting to split versions by platform, but I'm not sure we need that level of detail here.

I don't have any immediate preferences on colors. However, it would be good to be consistent across packages (e.g., osx-64 is always red, even if there aren't any linux packages).

And @epruesse I hope this means you are back in the game!?

File: src/build_plots/common.py (hunk 1 above)
Comment (aliciaaevans) [reply src/build_plots/common.py]: Thanks for the suggestions!

**Review by daler — APPROVED** [review]

**Comment by aliciaaevans** [comment]
@daler I updated the color scheme (and example in the PR description) to a paired palette with fixed colors for each platform. The only down side is that if we add more platforms we'll have to update the array, but that would have to be done across all the repos anyway.

**Pull request merged**